src/h3lib/lib/faceijk.c

```diff
@@ -891,3 +906,21 @@ int _adjustOverageClassII(FaceIJK* fijk, int res, int pentLeading4,
 
     return overage;
 }
+
+/**
+ * Adjusts a FaceIJK address for a pentagon vertex in a substrate grid in
+ * place so that the resulting cell address is relative to the correct
+ * icosahedral face.
+ *
+ * @param fijk The FaceIJK address of the cell.
+ * @param res The H3 resolution of the cell.
+ */
+int _adjustPentVertOverage(FaceIJK* fijk, int res) {
+    int pentLeading4 = 0;
+    int overage = _adjustOverageClassII(fijk, res, pentLeading4, 1);
+    while (overage == 2) {
```

Comment:
Note that I cleaned up the logic here slightly to a single `while` loop.

Reply:
`_adjustOverageClassII` might be a good candidate for an enum. (Not a required change for this PR)

Reply:
Good idea, added.